test/extension/branches.test.ts

```diff
@@ -0,0 +1,149 @@
+        const expectedBranches = [
+            {
+                gitRoot: '',
+                name: 'addBranchTests',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'part1FixStartup',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'master',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: true,
+            },
+            {
+                gitRoot: '',
+                name: 'addTests',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'jest',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'replace-webserver-with-postmessage',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'localBranch3',
+                remote: '',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'localBranch2',
+                remote: '',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'localBranch1',
+                remote: '',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'curvyGraphs',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+            {
+                gitRoot: '',
+                name: 'WIP',
+                remote: 'https://github.com/DonJayamanne/test_gitHistory.git',
+                remoteType: 2,
+                current: false,
+            },
+        ].map(item => ({
+            gitRoot: localPath,
+            name: item.name,
+        }));
+
+        const factory = serviceManager.get<IGitServiceFactory>(IGitServiceFactory);
+        const gitService = await factory.createGitService(Uri.file(localPath));
+
+        await changeBranch(repoPath, 'master');
+
+        const branches = (await gitService.getBranches()).map(item => ({
```

Comment:
More tests to come where we validate the remote and other attributes of the branches.